ioc_fanger/ioc_fanger.py

```diff
@@ -75,14 +76,12 @@ def defang(text):
     """Defang the indicators in the given text."""
     defanged_text = text
 
-    for mapping in defanging_mappings:
+    for mapping in defang_mappings:
 
         def _replace(matches):
             """Replace matches.groups(1) with mapping['replace']."""
-            try:
-                return matches.group(0).replace(matches.group(1), mapping["replace"])
-            except IndexError:
-                return matches.group(0).replace(mapping["find"], mapping["replace"])
+            # we assume every regex will match a group and that group(1) is the one to be replaced
+            return matches.group(0).replace(matches.group(1), mapping["replace"])
 
         defanged_text = re.sub(mapping["find"], _replace, defanged_text)
```

Comment:
I think this could be updated to something like:

```suggestion
        defanged_text = mapping["find"].sub(_replace, defanged_text)
```